### PR TITLE
Feature/jedi gdas gnssro hailingz

### PR DESCRIPTION
**Pull request by hailingz**
This PR is to incorporate JCSDA's changes of GSI codes needed to create observations and geovals for JEDI nc_diag files.
Base branch is:
https://github.com/JCSDA-internal/GSI/tree/feature/jedi_gdas

The merge of this PR will close https://github.com/JCSDA-internal/ufo/issues/978

### CODE REVIEW

**Comment by hailingz** [comment]
Hi @fcvdb 

This is the PR for our GNSSRO ncdiag changes. Could you please take a look to see if the obs 1 label is correct? feel free to edit. Thanks,

**Review by fcvdb — APPROVED** [review]
Thanks @hailingz

File: src/gsi/genstats_gps.f90 [hunk 1]
```diff
@@ -752,41 +760,67 @@ subroutine contents_binary_diag_
...
+           refid    = gps_allptr%rdiag(25)
+           ogce     = gps_allptr%rdiag(28)
+           call nc_diag_metadata("reference_sat_id",            refid)
+           call nc_diag_metadata("occulting_sat_id@MetaData",   said)      ! LEO satellite ID
+           call nc_diag_metadata("occulting_sat_is@MetaData",   siid)      ! LEO satellite instrument
+           call nc_diag_metadata("gnss_sat_class@MetaData",     sclf)
+           call nc_diag_metadata("process_center@MetaData",     ogce)
+           call nc_diag_metadata("record_number@MetaData",      profid)  
+           call nc_diag_metadata("ascending_flag@MetaData",     ascd)
+           call nc_diag_metadata("latitude@MetaData",           sngl(gps_allptr%rdiag(3)) )
```
Comment (CoryMartin-NOAA) [inline src/gsi/genstats_gps.f90:793]: Are all these correct? Do these not run through the ioda-converters?

**Review by CoryMartin-NOAA — APPROVED** [review]
After speaking with @fcvdb this morning, seems my comment/question was addressed, these nc-diags are used directly, so I'll approve.

**Pull request merged**